### PR TITLE
fix environment variables deletion while updating a component

### PR DESCRIPTION
**Pull request by karthikjeeyar**
## Fixes 

https://issues.redhat.com/browse/HAC-2959

## Description

Deleting environment variables from UI was not reflecting in the components CR.

## Type of change

- [x] Bugfix

## Screen shots / Gifs for design review 

https://user-images.githubusercontent.com/9964343/213451990-367c5587-13fb-4e97-bbe4-5fbcf01e6457.mov

## Unit tests

```
Create Utils
    ✓ Should delete the environment variables while updating (1 ms)
    ✓ Should update the environment variables with new values (1 ms)
```

## How to test or reproduce?

1. While creating components, Add environment variables under advanced deployment options 
2. Go to component setting page from component list view and remove all the environment variables under advanced options.

## Browser conformance: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

cc: @christianvogt

### CODE REVIEW

File: src/utils/create-utils.ts [hunk 1]
```diff
@@ -114,7 +114,12 @@ export const createComponent = (
   };
 
   const resource =
-    verb === 'update' ? merge({}, originalComponent || {}, newComponent) : newComponent;
+    verb === 'update'
+      ? {
+          ...merge({}, originalComponent || {}, newComponent),
+          spec: newComponent.spec,
```
Comment (sahil143) [inline src/utils/create-utils.ts:120]: @karthikjeeyar This will completely replace the spec of the original component instead of a partial update.

Comment (christianvogt) [reply src/utils/create-utils.ts:120]: I think that's fair.
Why not just do:
```
  const resource =
    verb === 'update' ? { ...originalComponent, spec: newComponent.spec } : newComponent;
```

Comment (karthikjeeyar) [reply src/utils/create-utils.ts]: Updated the code.

**Review by rohitkrai03 — APPROVED** [review]
/lgtm

**Pull request merged**